### PR TITLE
Make fallback host files work correctly

### PR DESCRIPTION
**Pull request by seancpeters**

### CODE REVIEW

File: src/Microsoft.TemplateEngine.Edge/Settings/SettingsLoader.cs [hunk 1]
```diff
@@ -212,8 +194,30 @@ public ITemplate LoadTemplate(ITemplateInfo info)
 
         public IFile HostTemplateConfigFile(IFileSystemInfo config)
         {
-            string hostTemplateFileName = string.Join(string.Empty, _environmentSettings.Host.HostIdentifier, HostTemplateFileConfigBaseName);
-            return config.Parent.EnumerateFiles(hostTemplateFileName, SearchOption.TopDirectoryOnly).FirstOrDefault();
+            IDictionary<string, IFile> allHostFilesForTemplate = new Dictionary<string, IFile>();
+
+            foreach (IFile hostFile in config.Parent.EnumerateFiles(HostTemplateFileConfigBaseName, SearchOption.TopDirectoryOnly))
```
Comment (prafullbhosale) [inline src/Microsoft.TemplateEngine.Edge/Settings/SettingsLoader.cs]: Should the search pattern here be ` $"*{HostTemplateFileConfigBaseName}"`

File: src/Microsoft.TemplateEngine.Edge/Settings/SettingsLoader.cs [hunk 2]
```diff
@@ -212,8 +194,30 @@ public ITemplate LoadTemplate(ITemplateInfo info)
 
         public IFile HostTemplateConfigFile(IFileSystemInfo config)
```
Comment (mlorbetske) [inline src/Microsoft.TemplateEngine.Edge/Settings/SettingsLoader.cs]: Just noticed this (even though it's not part of this change) - method names should indicate actions

File: src/Microsoft.TemplateEngine.Edge/Settings/SettingsLoader.cs [hunk 3]
```diff
@@ -212,8 +194,30 @@ public ITemplate LoadTemplate(ITemplateInfo info)
...
-            string hostTemplateFileName = string.Join(string.Empty, _environmentSettings.Host.HostIdentifier, HostTemplateFileConfigBaseName);
-            return config.Parent.EnumerateFiles(hostTemplateFileName, SearchOption.TopDirectoryOnly).FirstOrDefault();
+            IDictionary<string, IFile> allHostFilesForTemplate = new Dictionary<string, IFile>();
+
+            foreach (IFile hostFile in config.Parent.EnumerateFiles(HostTemplateFileConfigBaseName, SearchOption.TopDirectoryOnly))
+            {
+                allHostFilesForTemplate.Add(hostFile.Name, hostFile);
+            }
+
+            string preferredHostFileName = string.Join(string.Empty, _environmentSettings.Host.HostIdentifier, HostTemplateFileConfigBaseName);
```
Comment (mlorbetske) [inline src/Microsoft.TemplateEngine.Edge/Settings/SettingsLoader.cs]: string.Concat(_environmentSettings.Host.HostIdentifier, HostTemplateFileConfigBaseName)

File: src/Microsoft.TemplateEngine.Edge/Settings/SettingsLoader.cs [hunk 4]
```diff
@@ -212,8 +194,30 @@ public ITemplate LoadTemplate(ITemplateInfo info)
...
+            if (allHostFilesForTemplate.TryGetValue(preferredHostFileName, out IFile preferredHostFile))
+            {
+                return preferredHostFile;
+            }
+
+            foreach (string fallbackHostName in _environmentSettings.Host.FallbackHostTemplateConfigNames)
+            {
+                string fallbackHostFileName = string.Join(string.Empty, fallbackHostName, HostTemplateFileConfigBaseName);
```
Comment (mlorbetske) [inline src/Microsoft.TemplateEngine.Edge/Settings/SettingsLoader.cs]: Also `Concat`

File: src/Microsoft.TemplateEngine.Edge/Settings/SettingsLoader.cs (hunk 1 above)
Comment (seancpeters) [inline src/Microsoft.TemplateEngine.Edge/Settings/SettingsLoader.cs]: It worked without it, but you're right, there is a file matching bug.
